cmd/bosun/sched/template.go

```diff
@@ -200,15 +188,50 @@ func (c *Context) eval(v interface{}, filter bool, series bool, autods int) (exp
 		}
 	}
 	if series && e.Root.Return() != parse.TypeSeries {
-		return nil, "", fmt.Errorf("egraph: requires an expression that returns a series")
+		return nil, "", fmt.Errorf("need a series, got %T (%v)", e, e)
 	}
 	res, _, err := e.Execute(c.runHistory.Context, c.runHistory.GraphiteContext, c.schedule.Conf.LogstashElasticHost, nil, c.runHistory.Start, autods, c.Alert.UnjoinedOK, c.schedule.Search, c.schedule.Conf.AlertSquelched(c.Alert))
 	if err != nil {
-		return nil, "", fmt.Errorf("%s: %v", v, err)
+		return nil, "", fmt.Errorf("%s: %v", e, err)
 	}
 	return res.Results, e.String(), nil
 }
 
+func (c *Context) evalResult(r expr.ResultSlice, filter bool, series bool, autods int) (expr.ResultSlice, string, error) {
+	if filter {
```

Comment:
The filter and series checks should just be done in eval so they are the same between evalResult and evalExpr. I realize we can optimize them, but I'd prefer the consistency instead.